### ISSUE TITLE
getMonthView() behaves incorrectly when excludedDays is set.

### ISSUE THREAD

**Issue by darrendlin** [mattlewis92/angular-calendar#1201]
**Describe the bug**

I'm trying to modify the MonthView to only display 2 weeks at a time. I'm attempting this with:
```
@Injectable()
export class HalfMonthCalendarUtils extends CalendarUtils {
  getMonthView(args: GetMonthViewArgs): MonthView {
    return super.getMonthView({
      ...args,
      viewStart: startOfWeek(args.viewDate),
      viewEnd: addWeeks(endOfWeek(args.viewDate), 1)
    });
  }
}
```
The above code, if I understand it correctly, should always display current and following week given a viewDate.

However, I noticed that if I have [excludeDays] set and I select a day in the last week of a month, then only the current week is displayed and the following week isn't displayed.

**Minimal reproduction of the problem with instructions**
- Have a basic CalendarUtils extension and modify the viewStart and viewEnd
- Have [excludeDays] set in the calendar MonthView
- It seems like if viewStart and viewEnd attempts to include days not within the current month, then those days get completely excluded even if the days should not have been filtered by [excludeDays].

https://stackblitz.com/edit/angular-kgs6vy

**Screenshots**
Expected:
<img width="1033" alt="Screen Shot 2020-03-22 at 5 00 02 PM" src="https://user-images.githubusercontent.com/3653692/77260504-c8371480-6c5e-11ea-824f-9219edb2b557.png">
Result:
<img width="983" alt="Screen Shot 2020-03-22 at 5 03 36 PM" src="https://user-images.githubusercontent.com/3653692/77260555-1b10cc00-6c5f-11ea-88f0-0b57919c02ef.png">

**Versions**

- `@angular/core`: 8.2.14
- `angular-calendar`: 0.28.5
- Browser name and version: Chrome 80.0.3987.132

**Comment by darrendlin** [comment]
After digging around in CalendarUtils, I think I found the source of the issue:
 https://github.com/mattlewis92/calendar-utils/blob/master/src/calendar-utils.ts#L846
```
...
      const isRowInMonth: boolean = row.some(
        day => getMonth(day.date) === getMonth(viewDate)
      );
      if (isRowInMonth) {
        days = [...days, ...row];
...
```

In instances where there are `excludeDays` set, the function only adds in the days if the date is in the same month. I'm running on fumes at the moment so I can't think of why it needs to check month as opposed to checking if it's within the given `viewStart` and `viewEnd`?

**Comment by mattlewis92** [comment]
Hey there, thanks for taking the time to look into this, I had a quick look and this certainly seems like a bug, would you like to [make a pull request](http://makeapullrequest.com/) with the fix you suggested? Thanks! 😄

**Comment by mattlewis92** [comment]
Should be fixed now in `0.28.8` 😄

**Comment by chitgoks** [comment]
If the option [weekStartsOn]="1", this previous week is displayed
